lib/nodes/LatchedSwitch.ts

```diff
@@ -0,0 +1,79 @@
+import {BlueshellState, ResultCode, isResultCode, rc, BaseNode, Conditional} from '../models';
+import {Parent} from './Parent';
+import {Constant} from './Constant';
+import {SwitchEntry} from './Switch';
+
+interface AdaptedSwitchEntry<S extends BlueshellState, E> {
+	conditional: Conditional<S, E>,
+	child: BaseNode<S, E>,
+}
+
+/**
+ * Executes the child associated with the first matching conditional.
+ * Returns defaultResult if no conditionals match.
+ *
```

Comment:
Comments here should mention that we only check the matching conditional once and then we latch until the child returns not RUNNING (something similar to the comment in LatchedIfElse).